spec/ad.ts

```diff
@@ -0,0 +1,430 @@
+        });
+        ads[0].should.have.property('year').equal(2010);
+      });
+      it('should be works with only max year value', async () => {
+        const mark = await Mark.findOne();
+        const ads = await AdService.getAdsByFilter({
+          markId: mark._id.toString(),
+          yearTo: 2017
+        });
+        ads[0].should.have.property('year').equal(2010);
+      });
+      it('should be works with only min year value', async () => {
+        const mark = await Mark.findOne();
+        const ads = await AdService.getAdsByFilter({
+          markId: mark._id.toString(),
+          yearFrom: 2008
+        });
+        ads[0].should.have.property('year').equal(2010);
+      });
+    });
+
+    describe('requests', () => {
+      it('should be return array of ads', async () => {
+        const res = await chai
+          .request(app)
+          .post('/api/ad')
+          .set('content-type', 'application/json')
+          .send({
+            filter: {
+              kmsFrom: 1000,
+              markId
+            },
+            limit: 2
+          });
+        const ads = res.body;
+        res.should.have.status(HttpStatus.OK);
+        ads.should.have.lengthOf(2);
+      });
+      it('should be return failed status if an internal db error occured', async () => {
+        const mongoStub = sinon
+          .stub(mongoose.Model, 'aggregate')
+          .returns((callback: any) => callback('Error'));
+        try {
+          await chai
+            .request(app)
+            .post('/api/ad')
+            .set('content-type', 'application/json')
+            .send({
+              filter: {
+                kmsFrom: 1000,
+                markId
+              }
+            });
+          chai.assert.fail('Test failed');
+        } catch (err) {
+          err.response.should.have.status(HttpStatus.INTERNAL_SERVER_ERROR);
+          chai.assert.equal(codeErrors.INTERNAL_DB_ERROR, err.response.body.error.code);
+        }
+        mongoStub.restore();
+      });
+      describe('Should be return failed status', () => {
+        it('if do not send required field', async () => {
+          try {
+            await chai
+              .request(app)
+              .post('/api/ad')
+              .set('content-type', 'application/json')
+              .send({});
+            chai.assert.fail('Test failed');
+          } catch (err) {
+            err.response.should.have.status(HttpStatus.UNPROCESSABLE_ENTITY);
+            chai.assert.equal(codeErrors.REQUIRED_FIELD, err.response.body.error.code);
+          }
+        });
+        it('if limit field is invalid', async () => {
+          try {
+            await chai
+              .request(app)
+              .post('/api/ad')
+              .set('content-type', 'application/json')
+              .send({
+                filter: {
+                  markId
+                },
+                limit: -1
+              });
+            chai.assert.fail('Test failed');
```

Comment:
What exactly's happened? Please use the descriptive error messages.
Something like "No error has been thrown by METHOD while the parameters: A, B were incorrect". (just an example)